src/snowflake/snowpark/session.py

```diff
@@ -2355,19 +2355,24 @@ def sproc(self) -> StoredProcedureRegistration:
         """
         return self._sp_registration
 
-    def _infer_is_return_table(self, sproc_name: str, *args: Any) -> bool:
+    def _infer_is_return_table(
+        self, sproc_name: str, *args: Any, log_on_exception: bool = False
+    ) -> bool:
         try:
             arg_types = [convert_sp_to_sf_type(infer_type(arg)) for arg in args]
             func_signature = f"{sproc_name.upper()}({', '.join(arg_types)})"
 
             # describe procedure returns two column table with columns - property and value
             # the second row in the sproc_desc is property=returns and value=<return type of procedure>
             # when no procedure of the signature is found, SQL exception is raised
-            sproc_desc = self._run_query(f"describe procedure {func_signature}")
+            sproc_desc = self._run_query(
+                f"describe procedure {func_signature}",
+                log_on_exception=log_on_exception,
+            )
             return_type = sproc_desc[1][1]
             return return_type.upper().startswith("TABLE")
         except Exception as exc:
-            _logger.warn(
+            _logger.info(
```

Comment:
Should this be guard by `log_on_exception`? I have a mixed feeling because I think this is actually a very useful log, so I prefer to keep it, but just want to raise it.

Reply:
The main concern here was if the describe procedure fails to find procedure, users see warning message even when things work successfully. There were two sources of logs, one here and one inside `_run_query`. 
Now the log message in `_run_query` is controlled by `log_on_exception`. I though keeping this one uniformly log info should be good. In this case, users can have messages generated at warning level when they want, else message go log info.